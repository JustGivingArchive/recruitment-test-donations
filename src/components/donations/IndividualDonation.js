import React from 'react';

const IndividualDonation = ({amount, message, imageUrl}) => {
  return (
    <div className="panel">
      <div className="donorinfo">
      <img
        className="donorimage"
        alt="donor profile picture"
        src={imageUrl}></img>
      <p className="paneltextlarge">Â£{amount || '--'}</p>
      </div>
      <p className="paneltext donormessage">{message}</p>
    </div>
  );
};

export default IndividualDonation;
