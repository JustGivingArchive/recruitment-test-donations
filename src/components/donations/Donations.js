import React from 'react';
import IndividualDonation from './IndividualDonation.js';

// donations are served from the API as an array, so we can map over them
// and create a donation component for each

const Donations = ({ donations }) => {
  return (
    <div className="content_container">
      {donations.map((data, key) => <IndividualDonation
        key={key}
        {...data} />)}
    </div>
  );
};

export default Donations;
