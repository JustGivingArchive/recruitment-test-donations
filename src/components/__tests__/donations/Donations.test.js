import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from "enzyme";

import Donations from '../../donations/Donations.js';
import IndividualDonation from '../../donations/IndividualDonation.js';

describe('Test donation info componenet', () => {
  let props;
  let mounted;
  const donations = () => {
    if (!mounted) {
      mounted = mount(
        <Donations {...props} />
      );
    }
    return mounted;
  }

  beforeEach(() => {
    props = {
    };
    mounted = undefined;
  });

  it('Matches snapshop correctly', () => {
    const tree = renderer.create(
      <Donations />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Always renders a containing div', () => {
    const divs = donations().find("div");
    expect(divs.length).toBeGreaterThan(0);
  });

  it("wrapping div contains everything", () => {
    const divs = donations().find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(donations().children());
  });

});
