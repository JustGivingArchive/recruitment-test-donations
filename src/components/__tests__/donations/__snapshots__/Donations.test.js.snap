exports[`Test donation info componenet Matches snapshop correctly 1`] = `
<div
  className="content_container">
  <div
    className="panel">
    <div
      className="donorinfo">
      <img
        alt="donor profile picture"
        className="donorimage"
        src="http://image.jpeg" />
      <p
        className="paneltextlarge">
        Â£
        5
      </p>
    </div>
    <p
      className="paneltext donormessage">
      hello
    </p>
  </div>
</div>
`;
